src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        GeometricFigure figure = new GeometricFigureSupplier().getRandomFigure();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        if (figure == GeometricFigure.CIRCLE) {
+            return new Circle(random.nextInt(), colorSupplier.getRandomColor().toLowerCase());
+        }
+        if (figure == GeometricFigure.RECTANGLE) {
+            return new Rectangle(random.nextInt(), random.nextInt(),
+                    colorSupplier.getRandomColor().toLowerCase());
+        }
+        if (figure == GeometricFigure.ISOSCELES_TRAPEZOID) {
+            return new IsoscelesTrapezoid(random.nextInt(), random.nextInt(),
+                    random.nextInt(), colorSupplier.getRandomColor().toLowerCase());
+        }
+        if (figure == GeometricFigure.SQUARE) {
+            return new Square(random.nextInt(), colorSupplier.getRandomColor().toLowerCase());
+        }
+        return new RightTriangle(random.nextInt(), random.nextInt(),
+                colorSupplier.getRandomColor().toLowerCase());
```

Comment:
You should create several random Figures, so you will use a loop. Please don't create a 'new FigureSupplier()' inside the loop. Also, consider using a switch-case structure instead of multiple if-statements to handle different 'GeometricFigure' values.